### PR TITLE
[bitnami/mongodb] Add MongoDB container VIB tests

### PR DESCRIPTION
**Pull request by mdhont**
The main objective of this PR is to publish our Bitnami MongoDB container using VMware Image Builder. In order to do that, several changes are included:

Increasing the existing test coverage of the asset by adding Goss tests.
Update verify and publish VIB pipeline's definitions.

### Benefits

Ensuring higher quality of the container catalog.
Increased pool of assets completely handled by VMware Image Builder.

### Possible drawbacks

Automated tests could introduce additional flakiness to the CI/CD.

### CODE REVIEW

File: .vib/mongodb/goss/mongodb.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+file:
```
Comment (joancafom) [inline .vib/mongodb/goss/mongodb.yaml:1]: You should check `/opt/bitnami/mongodb/conf/mongodb.conf` and some of its rendered content

**Review by joancafom — CHANGES_REQUESTED** [review]
Thanks for your PR! I left you some comments as a review

**Review by FraPazGal — CHANGES_REQUESTED** [review]
Hi @mdhont, please review my comments when you can. There is a couple of things I have concerns about.

File: .vib/mongodb/goss/vars.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+binaries:
+  - gosu
+  - mongod
+  - mongosh
+  - render-template
+  - wait-for-port
+  - yq
+files:
+  - paths:
```
Comment (joancafom) [inline .vib/mongodb/goss/vars.yaml:9]: You are not checking the permissions

File: .vib/mongodb/goss/vars.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    - /.dbshell
+    - /.mongoshrc.js
+    - /.mongorc.js
+    - /opt/bitnami/mongodb/conf/mongodb.conf
```
Comment (joancafom) [inline .vib/mongodb/goss/vars.yaml:13]: This is already checked in the `mongodb.yaml` file

File: bitnami/mongodb/5.0/debian-11/Dockerfile [hunk 4]
```diff
@@ -18,7 +18,7 @@ ENV HOME="/" \
 
 COPY prebuildfs /
 SHELL ["/bin/bash", "-o", "pipefail", "-c"]
-# Install required system packages and dependencies
+## Install required system packages and dependencies
```
Comment (joancafom) [inline bitnami/mongodb/5.0/debian-11/Dockerfile:21]: Remove the trigger files once the PR is marked as ready for review

File: .vib/mongodb/goss/mongodb.yaml [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    contains:
+      - This solution ships the official MongoDB Community binaries
+  /opt/bitnami/mongodb/logs/mongodb.log:
+    exists: true
+    filetype: symlink
+    linked-to: /dev/stdout
+  /opt/bitnami/mongodb/conf/mongodb.conf:
+    exists: true
+    contains:
+      - /port.*{{ .Vars.port }}/
```
Comment (joancafom) [inline .vib/mongodb/goss/mongodb.yaml:13]: Don't use vars here, use the raw value instead as it is unlikely to change.

File: .vib/mongodb/goss/mongodb.yaml [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+      - /pidFilePath.*opt\/bitnami\/mongodb\/tmp\/mongodb.pid/
+      - /path.*\/opt\/bitnami\/mongodb\/logs\/mongodb.log/
```
Comment (joancafom) [inline .vib/mongodb/goss/mongodb.yaml:15]: The `/` character does not need to be escaped

File: .vib/mongodb/goss/vars.yaml [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+  - mode: "0775"
+    paths:
+      - /bitnami/mongodb
+      - /bitnami/mongodb/data
+      - /docker-entrypoint-initdb.d
+      - /opt/bitnami/mongodb/conf
+      - /opt/bitnami/mongodb/logs
+      - /opt/bitnami/mongodb/tmp
+      - /.mongodb
+port: 27017
```
Comment (joancafom) [inline .vib/mongodb/goss/vars.yaml:24]: Remove this line

**Review by joancafom — CHANGES_REQUESTED** [review]

File: .vib/mongodb/goss/mongodb.yaml [hunk 8]
```diff
@@ -0,0 +1,17 @@
+file:
+  /opt/bitnami/mongodb/mongodb_get_source.txt:
+    mode: "0644"
+    exists: true
+    contains:
+      - This solution ships the official MongoDB Community binaries
```
Comment (FraPazGal) [inline .vib/mongodb/goss/mongodb.yaml:6]: Minor comment, I think checking for such a long string is very error-prone if upstream decides to change some of it.

```suggestion
      - official MongoDB Community
```

File: .vib/mongodb/goss/mongodb.yaml [hunk 9]
```diff
@@ -0,0 +1,17 @@
+file:
+  /opt/bitnami/mongodb/mongodb_get_source.txt:
+    mode: "0644"
+    exists: true
+    contains:
+      - official MongoDB Community
```
Comment (joancafom) [inline .vib/mongodb/goss/mongodb.yaml:6]: Please, add back the whole string again. This is not something we get from upstream, but we add it ourselves

**Review by joancafom — APPROVED** [review]
LGTM!

**Review by FraPazGal — APPROVED** [review]
LGTM!

**Pull request merged**